### PR TITLE
No space in by-name params

### PR DESCRIPTION
**Pull request by vovapolu**
Fixing https://github.com/scalameta/scalafmt/issues/1067

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/config/Spaces.scala [hunk 1]
```diff
@@ -26,5 +29,6 @@ case class Spaces(
     inImportCurlyBraces: Boolean = false,
     inParentheses: Boolean = false,
     neverAroundInfixTypes: Seq[String] = Nil,
-    afterKeywordBeforeParen: Boolean = true
+    afterKeywordBeforeParen: Boolean = true,
+    noSpaceInByNameParam: Boolean = false
```
Comment (olafurpg) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/config/Spaces.scala]: `inByNameTypes: Boolean = true`

File: scalafmt-tests/src/test/resources/spaces/NoSpaceInByNameParam.stat [hunk 2]
```diff
@@ -0,0 +1,21 @@
+spaces.noSpaceInByNameParam = true
+<<< No space in by-name parameter
+def foo(a: => A) = ???
+>>>
+def foo(a: =>A) = ???
+<<< Space in function parameter
+def foo(a: B => A) = ???
+>>>
+def foo(a: B => A) = ???
```
Comment (olafurpg) [inline scalafmt-tests/src/test/resources/spaces/NoSpaceInByNameParam.stat]: good to test this 👍

**Comment by fommil** [comment]
omg, I want this yesterday!

**Comment by olafurpg** [comment]
You can try this out today, the CI publishes artifacts on merge 😄 

```
coursier fetch com.geirsson:scalafmt-core_2.12:1.3.0+14-910a2ed5 -r bintray:scalameta/maven
```

**Pull request merged**